zgw/common-ground-authentication/src/main/kotlin/nl/nlportal/commonground/authentication/CommonGroundAuthenticationConverter.kt

```diff
@@ -15,43 +15,87 @@
  */
 package nl.nlportal.commonground.authentication
 
-import nl.nlportal.commonground.authentication.exception.UserTypeUnsupportedException
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonValue
 import mu.KotlinLogging
+import nl.nlportal.commonground.authentication.exception.UserTypeUnsupportedException
 import org.springframework.core.convert.converter.Converter
+import org.springframework.core.env.Environment
 import org.springframework.security.oauth2.jwt.Jwt
+import org.springframework.security.oauth2.jwt.ReactiveJwtDecoder
 import org.springframework.security.oauth2.server.resource.authentication.JwtGrantedAuthoritiesConverter
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
 import reactor.core.publisher.Mono
+import java.net.URI
+
 
-class CommonGroundAuthenticationConverter : Converter<Jwt, Mono<CommonGroundAuthentication>> {
+class CommonGroundAuthenticationConverter(val environment: Environment, val decoder: ReactiveJwtDecoder) : Converter<Jwt, Mono<CommonGroundAuthentication>> {
```

Comment:
Waarom wordt dit opgelost met environment? en niet correct via een ConfigurationProperties met prefix = keycloak.  Het gebruik van environment is afwijkend van de manier hoe andere properties op de Spring manier ingeladen worden

Reply:
En graag de tests fixen, er zijn er die falen

Reply:
Test kan ik niet locaal runnen dus daar moet ik nog even met peter naar kijken het andere doe ik

Reply:
testen run ik niet in IntelliJ, maar buiten IntelliJ via een command line met gradle build en dat gaat best goed ;-)